tools/Azure.Mcp.Tools.Communication/src/Services/CommunicationService.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Communication.Sms;
+using Azure.Core;
+using Azure.Mcp.Core.Options;
+using Azure.Mcp.Core.Services.Azure;
+using Azure.Mcp.Core.Services.Azure.Subscription;
+using Azure.Mcp.Core.Services.Azure.Tenant;
+using Azure.Mcp.Tools.Communication.Models;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Mcp.Tools.Communication.Services;
+
+public class CommunicationService(ISubscriptionService subscriptionService, ITenantService tenantService, ILogger<CommunicationService> logger)
+    : BaseAzureService(tenantService), ICommunicationService
+{
+    private readonly ISubscriptionService _subscriptionService = subscriptionService ?? throw new ArgumentNullException(nameof(subscriptionService));
+    private readonly ILogger<CommunicationService> _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+    public async Task<List<SmsResult>> SendSmsAsync(
+        string endpoint,
+        string from,
+        string[] to,
+        string message,
+        bool enableDeliveryReport = false,
+        string? tag = null,
+        string? tenantId = null,
+        RetryPolicyOptions? retryPolicy = null)
+    {
+        // Validate required parameters using base class method
+        ValidateRequiredParameters(
+            (nameof(endpoint), endpoint),
+            (nameof(from), from),
+            (nameof(message), message));
+
+        // Validate to array separately since it has special requirements
+        if (to == null || to.Length == 0)
+            throw new ArgumentException("At least one 'to' phone number must be provided", nameof(to));
+
+        if (to.Any(string.IsNullOrWhiteSpace))
+            throw new ArgumentException("All To phone numbers must be valid", nameof(to));
```

Comment:
You're not validating the phone numbers here, instead the error message should be something like:


```suggestion
            throw new ArgumentException("Recipient phone numbers cannot be empty.", nameof(to));
```